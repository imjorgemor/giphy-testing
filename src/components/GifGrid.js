import { useFetchGifs } from "../hooks/useFetchGifs"
import GifGridItem from "./GifGridItem";
import PropTypes from 'prop-types'

const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{category === "trending" ? null : category}</h3>

            {category.length > 1 && loading && <p>loading...</p>}

            <div className="card-grid">
                {
                    data.map(img => (
                        <GifGridItem key={img.id} {...img} />
                    ))
                }
            </div>
        </>
    )
}

export default GifGrid

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}