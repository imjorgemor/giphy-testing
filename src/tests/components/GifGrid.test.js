import { shallow } from "enzyme";
import "@testing-library/jest-dom"
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("test sobre <GifGrid />", () => {

    const category = "simpsons"


    test("test de renderizado, snapshot", () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const component = shallow(<GifGrid category={category} />)
        expect(component).toMatchSnapshot();
    })

    test("debe mostrar items cuando se cargan imagenes con el useFetch", () => {

        const gifs = [{
            id: "test",
            url: "https://localhost/test.jpg",
            title: "test"
        },
        {
            id: "test1",
            url: "https://localhost/test.jpg",
            title: "test"
        }
        ]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const component = shallow(<GifGrid category={category} />)
        //comprobamos si el loading existe
        //expect(component).toMatchSnapshot();
        expect(component.find('p').exists()).toBe(false)
        expect(component.find('GifGridItem').length).toBe(gifs.length)

    })

})