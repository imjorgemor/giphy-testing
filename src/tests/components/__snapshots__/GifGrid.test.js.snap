// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test sobre <GifGrid /> test de renderizado, snapshot 1`] = `
<Fragment>
  <h3>
    simpsons
  </h3>
  <p>
    loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
